src/pathfinding_service/service.py

```diff
@@ -168,26 +169,66 @@ def _process_new_blocks(self, to_block: BlockNumber) -> None:
-        _, events = get_blockchain_events(
-            web3=self.web3,
-            contract_manager=CONTRACT_MANAGER,
-            chain_state=self.blockchain_state,
-            to_block=to_block,
+        # increment by one, as `latest_committed_block` has been queried last time already
+        from_block = BlockNumber(self.blockchain_state.latest_committed_block + 1)
+        to_block = min(
+            latest_confirmed_block,
+            # decrement by one, as `from_block` is inclusive
+            BlockNumber(from_block + self.blockchain_state.current_event_filter_interval - 1),
         )
-        before_process = time.time()
+
+        try:
+            before_query = time.monotonic()
+            _, events = get_blockchain_events(
+                web3=self.web3,
+                contract_manager=CONTRACT_MANAGER,
+                chain_state=self.blockchain_state,
+                from_block=from_block,
+                to_block=to_block,
+            )
+            after_query = time.monotonic()
+
+            filter_query_duration = after_query - before_query
+            if filter_query_duration < ETH_GET_LOGS_THRESHOLD_FAST:
+                self.blockchain_state.current_event_filter_interval = BlockTimeout(
+                    min(
+                        MAX_FILTER_INTERVAL,
+                        self.blockchain_state.current_event_filter_interval * 2,
+                    )
+                )
+            elif filter_query_duration > ETH_GET_LOGS_THRESHOLD_SLOW:
+                self.blockchain_state.current_event_filter_interval = BlockTimeout(
+                    max(
+                        MIN_FILTER_INTERVAL,
+                        self.blockchain_state.current_event_filter_interval // 2,
+                    )
+                )
+        except ReadTimeout:
+            old_interval = self.blockchain_state.current_event_filter_interval
+            self.blockchain_state.current_event_filter_interval = BlockTimeout(
+                max(MIN_FILTER_INTERVAL, old_interval // 5)
+            )
+            log.debug(
+                "Failed to query events in time, reducing interval",
+                old_interval=old_interval,
+                new_interval=self.blockchain_state.current_event_filter_interval,
+            )
+            return
+
+        before_process = time.monotonic()
         for event in events:
             self.handle_event(event)
             gevent.idle()  # Allow answering requests in between events
 
         if events:
             log.info(
                 "Processed blocks",
-                from_block=self.blockchain_state.latest_committed_block + 1,
+                from_block=from_block,
                 to_block=to_block,
-                getting=round(before_process - before_get, 2),
-                processing=round(time.time() - before_process, 2),
-                total_duration=round(time.time() - start, 2),
+                interval=to_block - from_block,
```

Comment:
you probably also want to do the `+1` here? https://github.com/raiden-network/raiden-services/pull/783/files#diff-15eaee2c72a6c3d020822b921b44d79aR167

Reply:
Ah, this is just a leftover. I'll remove it